components/mediation/data-publishers/org.wso2.micro.integrator.analytics.messageflow.data.publisher/src/test/java/org/wso2/micro/integrator/analytics/messageflow/data/publisher/publish/elasticsearch/ElasticStatisticsTest.java

```diff
@@ -272,11 +268,25 @@ private PublishingEvent createPublishingEvent(ComponentType componentType, Strin
         event.setDuration(event.getEndTime() - event.getStartTime());
         event.setEntryPoint("EP");
         event.setFaultCount(0);
-        ElasticMetadata elasticMetadata = new ElasticMetadata(messageContext);
-        event.setElasticMetadata(elasticMetadata);
+        event.setElasticMetadata(createElasticMetadata());
         return event;
     }
 
+    private ElasticMetadata createElasticMetadata() {
+        AnalyticsCustomDataProvider customDataProvider = new SampleCustomDataProvider();
+        Map<String, Object> customProperties = customDataProvider.getCustomProperties(messageContext);
+        Map<String, Object> contextProperties =  new HashMap<>(messageContext.getProperties());
+        contextProperties.computeIfAbsent(SynapseConstants.ANALYTICS_METADATA, k -> new HashMap<>());
+        ((HashMap<String, Object>) contextProperties.get(SynapseConstants.ANALYTICS_METADATA)).putAll(customProperties);
+        return new ElasticMetadata(
+                messageContext.getConfiguration(),
+                messageContext.isFaultResponse(),
+                messageContext.getMessageID(),
+                messageContext.getContextEntries(),
+                contextProperties
+        );
+    }
```

Comment:
@gayaldassanayake are we treating the `metadata` object as the "custom property bag"? Don't we have another map which maintains custom properties?

Reply:
We already have a map that contains the analytics properties passed by the analytics scope. That is contained within the contextProperties map.

i.e.
``` java
Map<String, Object> contextProperties = messageContext.getProperties(); 
Map<String, Object> analyticsMetadata = contextProperties.get(SynapseConstants.ANALYTICS_METADATA); 
// This is the map that already contains the analytics metadata. 
// Any custom properties passed by 'AnalyticsCustomDataProvider' are also populated into this map.
```